core/src/wallet/bitcoin/transaction_builders/BitcoinLikeStrategyUtxoPicker.cpp

```diff
@@ -212,44 +181,52 @@ namespace ledger {
             int64_t longTermFees = DEFAULT_FALLBACK_FEE;
 
             //Compute cost of change
-            auto fixedSize = BitcoinLikeTransactionApi::estimateSize(0,
+            auto const fixedSize = BitcoinLikeTransactionApi::estimateSize(0,
                                                                      0,
                                                                      currency,
                                                                      buddy->keychain->getKeychainEngine());
             //Size of only 1 output (without fixed size)
-            auto oneOutputSize = BitcoinLikeTransactionApi::estimateSize(0,
+            auto const oneOutputSize = BitcoinLikeTransactionApi::estimateSize(0,
                                                                          1,
                                                                          currency,
                                                                          buddy->keychain->getKeychainEngine()).Max - fixedSize.Max;
             //Size 1 signed UTXO (signed input)
-            auto signedUTXOSize = BitcoinLikeTransactionApi::estimateSize(1,
+            auto const signedUTXOSize = BitcoinLikeTransactionApi::estimateSize(1,
                                                                           0,
                                                                           currency,
                                                                           buddy->keychain->getKeychainEngine()).Max - fixedSize.Max;
 
             //Size of unsigned change
-            auto changeSize = oneOutputSize;
+            auto const changeSize = oneOutputSize;
             //Size of signed change
-            auto signedChangeSize = signedUTXOSize;
+            auto const signedChangeSize = signedUTXOSize;
 
-            auto effectiveFees = buddy->request.feePerByte;
+            auto const effectiveFees = buddy->request.feePerByte;
             // Here signedChangeSize should be multiplied by discard fees
             // but since we don't have access to estimateSmartFees, we assume
             // that discard fees are equal to effectiveFees
             const int64_t costOfChange = effectiveFees->toInt64() * (signedChangeSize + changeSize);
+
             buddy->logger->debug("Cost of change {}, signedChangeSize {}, changeSize {}", costOfChange, signedChangeSize, changeSize);
 
             //Calculate effective value of outputs
             int64_t currentAvailableValue = 0;
-            using EffectiveUTXOList = std::vector<EffectiveUTXO>;
-            EffectiveUTXOList listEffectiveUTXOs;
+
+            struct EffectiveUtxo {
+                const BitcoinLikeUtxo *utxo;
```

Comment:
Since the structure is for internal mechanism, I chose a pointer for convenience (I struggled a lot to compile with reference - usage of deleting copy constructor which should not happen...).
To conclude the pointer is never null and a better solution should probably find out why reference cannot be use or use `gsl::not_null`

Reply:
Yeah, C++ madness.